.github/workflows/test.yml

```diff
@@ -34,6 +34,14 @@ jobs:
       - name: Checkout
         uses: actions/checkout@v4
 
+      - name: Patch DNS
+        uses: cylc/release-actions/patch-dns@v1
+
+      - name: Setup Python
+        uses: actions/setup-python@v5
+        with:
+          python-version: ${{ matrix.python }}
```

Comment:
This was not actually failing because Python 3.9 was removed; it would install it anyway. But the installation was failing due to
```
chmod: changing permissions of 'python3.9': Operation not permitted
```
([example](https://github.com/cylc/cylc-uiserver/actions/runs/12067460876/job/33650461146#step:5:26))

For some reason this is fixed by moving to before the brew install step

Reply:
> For some reason this is fixed by moving to before the brew install step

That may suggest that the Brew Python is replacing the system Python?

Might be worth testing `python --version` at some point after this step to make sure we're actually getting the version of Python we are configuring?

Reply:
Pytest is reporting 3.9: https://github.com/cylc/cylc-uiserver/actions/runs/12146399589/job/33870231037?pr=650#step:11:14